### ISSUE TITLE
`address` in config not working

### ISSUE THREAD

**Issue by yifeikong** [hashicorp/consul#4404]
#### Overview of the Issue

When defining services in `/etc/consul.d`, address attribute of a service is not working.

#### Reproduction Steps

`/etc/consul.d/redis.json` :

```
{
    "services": [
        {
            "name": "redis.db",
            "tags": ["db"],
            "address": "10.1.1.2",
            "port": 6379
        }
    ]
}
```
consul command:

```
consul agent -server -ui -client=0.0.0.0 -data-dir=/var/consul -bind={{ GetInterfaceIP "eth0" }} -config-dir=/etc/consul.d/
```
after `consul reload`, the port part DID change, however, the address is always node's IP

### Consul info for both Client and Server

<details>
  <summary>Client info</summary>

```
output from client 'consul info' command here
```

</details>

<details>
  <summary>Server info</summary>

```
output from server 'consul info' command here
```

</details>

### Operating system and Environment details

Ubuntu 16.04, consul 1.2.1

### Log Fragments
```
Jul 17 07:10:44 ip-10-1-1-1 consul[12109]:     2018/07/17 07:10:44 [INFO] agent: Reloading configuration...
Jul 17 07:10:44 ip-10-1-1-1 consul[12109]:     2018/07/17 07:10:44 [WARN] agent: Service name "redis.db" will not be discoverable via DNS due to invalid characters. Valid characters
Jul 17 07:10:44 ip-10-1-1-1 consul[12109]:     2018/07/17 07:10:44 [INFO] agent: Synced service "redis.db"
```

**Comment by mkeeler** [comment]
@yifeikong What are you doing to determine the address has not changed? Is it a health check that isn't using the new address or another end point?

**Comment by yifeikong** [comment]
I define the service address as '10.1.1.2:6379' as in the `redis.json` file on host 10.1.1.1, however, in the consul ui, the service is alway marked as ‘10.1.1.1:6379’

**Comment by mkeeler** [comment]
@yifeikong If you `curl <agent address>/v1/catalog/service/redis.db` does it produce the correct address. If so this might be a caching issue in the UI.

**Comment by yifeikong** [comment]
No, the api also gives error

**Comment by johncowen** [comment]
Hi all,

It sounds like this isn't the issue if the API is not returning what you expect, but just incase, we did have an issue with showing service IP addresses in the UI where we were showing the Node IP instead of the Service IP (which sounds like what you describe here), this is fixed in #4410 

That fix will be in the next release.

Thanks,

**Comment by johncowen** [comment]
Hey @yifeikong ,

Thanks!

Just to finish up, if anyone else comes across this, the UI fix discussed above has been released and is in the 1.2.2 version of Consul.
